lilypad/server/main.py

```diff
@@ -206,6 +198,84 @@ async def get_llm_function_id_by_hash(
     return llm_function if llm_function else None
 
 
+def create_dynamic_function(
+    name: str, args_dict: dict[str, str], return_annotation: type
+) -> str:
+    """Create a dynamic function with the given name, arguments, and return type."""
+    arg_list = [f"{arg_name}: {arg_type}" for arg_name, arg_type in args_dict.items()]
+    arg_string = ", ".join(arg_list)
+
+    func_def = f"def {name}({arg_string}) -> {return_annotation.__name__}: ..."
+
+    namespace: dict[str, str] = {}
+    exec(func_def, namespace)
+
+    return namespace[name]
+
+
+@api.post("/projects/{project_id}/versions/{version_id}/vibe")
+async def vibe_check(
+    project_id: int,
+    version_id: int,
+    arg_values: dict[str, Any],
+    session: Annotated[Session, Depends(get_session)],
+) -> str:
+    """Check if the vibes are good."""
+    version_table = session.exec(
+        select(VersionTable).where(
+            VersionTable.project_id == project_id, VersionTable.id == version_id
+        )
+    ).first()
+    if not version_table:
+        raise HTTPException(status_code=404, detail="Version not found")
+    version = VersionPublic.model_validate(version_table)
+    args_dict = version.llm_fn.arg_types or {}
+    configure()
+    trace_decorator = trace(
+        project_id=project_id,
+        version_id=version.id,
+        arg_types=args_dict,
+        arg_values=arg_values,
+        lexical_closure=version.llm_fn.code,
+        prompt_template=version.fn_params.prompt_template if version.fn_params else "",
+        version=version.version,
+    )
+
+    fn = create_dynamic_function(
+        name=version.llm_fn.function_name,
+        args_dict=args_dict,
+        return_annotation=str,
+    )
+
+    if not version.fn_params:
+        return trace_decorator(fn)(**arg_values)
```

Comment:
should i just type ignore this or is there a way to get typing

Reply:
just use the specific `pyright: ignore [{SPECIFIC_ERROR}]` for now, we can go through type ignores later and fix them, but not super high priority right now so long as it works